src/main/java/com/olgaboiar/mint/Header.java

```diff
@@ -30,25 +31,59 @@ String getDate() {
         return date;
     }
 
-    String prepareHeaders() {
-        if (allowedMethods != null) {
-            String methods = String.join(",", allowedMethods);
-            String accessControl = "Allow: " + methods;
-            headersToSend = statusLine + "\n"
-                    + contentType + "\n"
-                    + date + "\n"
-                    + accessControl + "\n"
-                    + blankLine;
-        } else {
-            headersToSend = statusLine + "\n"
-                + contentType + "\n"
-                + date + "\n"
-                + blankLine;
+    public ArrayList<String> getHeaders() {
+        ArrayList<String> headers = new ArrayList<String>();
+        headers.add(getStatusLine());
+        headers.add(getContentType());
+        headers.add(getDate());
+        if (allowedMethodsExist()) {
+            String aloowHeader = createAllowHeader();
+            headers.add(aloowHeader);
+        }
+        if (redirectExist()) {
+            String redirectHeader = createRedirectHeader();
+            headers.add(redirectHeader);
         }
+        return headers;
+    }
+
+    public String prepareHeaders(ArrayList<String> headers) {
+        String headersToSend = String.join("\n", headers);
+        headersToSend += BLANK_LINE;
         return headersToSend;
     }
```

Comment:
It looks like the only time this `prepareHeaders` is used on the return of `getHeaders` method. I think it's probably OK to just call `getHeaders()` inside of the method body and have `prepareHeaders` not take an argument. As an aside, at that point it seems like really `prepareHeaders` would bet more appropriately named: `ToString()`.

Reply:
refactored in https://github.com/olgaboiar/MintHTTPServer/pull/9/commits/7309f1920c7cc80d41fc457c04302c56327bb501